### ISSUE TITLE
[Circle-OpSelector-Test] There is a problem in comparing two circles because of random parameters by recipe 

### ISSUE THREAD

**Issue by parkeunsang** [Samsung/ONE#7787]
We're developing `circle-opselector-test` (related pr : #7765, issue : #7671)
The test process proceeds as follows
1. Prepare `tflite recipe` for origin circle model(**origin.circle**). The recipe was already in `res/TensorFlowLiteRecipes` such as Part_Sqrt_Rsqrt_001
2. Select the nodes to be cut from the model(**origin.circle**) and make a `tflite recipe` consisting of only the nodes. Output file is **selected.tflite**
3. Select the node using `circle-opselector`. And output file is **selected.circle**
4. Compare two models(selected.tflite and selected.circle) using `luci-value-test` which compares the out value of models by random input value.

In this process, We saw a problem with models with parameters to learn, such as Conv2d. Recipe generates the parameters randomly. So **selected.tflite** and **selected.circle** have the same structure but have the different parameters so `luci-value-test`'s result is also failed.
Is there any solution to solve this problem? thanks

**Comment by seanshpark** [comment]
You need to check this as I've never tried this before.
- If you are working on a code copied/modified from `compiler/luci-value-test/luci_eval_verifier.py`,
- change `interpreter.get_tensor(interpreter.get_output_details()[idx]["index"])` to something else, where
- index = the node index that maps to selected.circle node index
- tensor = interpreter.get_tensor(index)
- I think you can get the shape from tensor.shape... but not sure

**Comment by parkeunsang** [comment]
> @seanshpark You need to check this as I've never tried this before.

I think this method is comparing only `structure of models`. But  I think that we should also check the parameters in operators which were selected by `circle-opselector`. Is this right?

**Comment by mhs4670go** [comment]
@parkeunsang 
>  have the same structure but have the different parameters so

I think the parameter you said may be weight or something.

Could you check the `tag` of `filler`? 
```flatbuffers
operand {
  name: "ker"
  type: FLOAT32
  shape { dim: 1 dim: 1 dim: 1 dim: 2 }
  filler {
    tag: "gaussian"
    arg: "0.0"
    arg: "1.0"
  }
}
operand {
  name: "bias"
  type: FLOAT32
  shape { dim: 1 }
  filler {
    tag: "constant"
    arg: "1.1"
  }
}
```

There are `gaussian`, `explicit` and `constant` that represents which policy will be applied when the wieghts are generated. If you choose `explicit`, you can fix the result of weight.

You can create fixed weight recipe with `tflchef-reverse` driver.

**Comment by glistening** [comment]
> I think the parameter you said may be weight or something.

Yes, @parkeunsang means conv2d weight tensor's values by `parameter` in above comments.

**Comment by parkeunsang** [comment]
@mhs4670go 
> There are gaussian, explicit and constant that represents which policy will be applied when the wieghts are generated. If you choose explicit, you can fix the result of weight.
You can create fixed weight recipe with tflchef-reverse driver.

Thanks. This solution will help a lot.

**Comment by glistening** [comment]
> You can create fixed weight recipe with `tflchef-reverse` driver.

@mhs4670go Could you point out the document or show example to specify `explicit`? I cannot find document. I am searching for corresponding source code.

**Comment by seanshpark** [comment]
> I think this method is comparing only structure of models

No, 

```py
            if np.allclose(
                    luci_output_data,
                    interpreter.get_tensor(
                        interpreter.get_output_details()[idx]["index"]),
                    rtol=0,
                    atol=0) == False:
                raise SystemExit("Execution result of " + tflite_model +
                                 " does not match with " + circle_model)
```
this code compares the execution tensor values.

**Comment by mhs4670go** [comment]
@glistening @parkeunsang 
> Could you point out the document or show example to specify `explicit`?

There's no document about this. T_T

> You can create fixed weight recipe with `tflchef-reverse` driver.

Well, FYI, after I checked the codes, I realized that `tflchef-reverse` doesn't support creating the recipe that has `explicit` filler. But, you can use `explicit` keyword with `tflchef-file` driver. It is a bit cumbersome to write `arg` yourself, but please understand about this:)

**Comment by seanshpark** [comment]
not sure to close or not. @mhs4670go any comments?
